src/functions-and-arrays.js

```diff
@@ -26,16 +61,48 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numArray) {
+
+if (numArray.length === 0){
+  return null;
+}
+let total = 0;
+
+for (const sumOfNumbers of numArray){
+  total += sumOfNumbers;
+}
+return total/numArray.length;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+
+function sumWordsLength(listOfWords) {   
+  let sumOfStrings = 0; 
+   for (i = 0; i < listOfWords.length; i++) {    
+    sumOfStrings += listOfWords[i].length;
+     }   
+     return sumOfStrings;
+    } 
+  
+    function averageWordLength(listOfWords) {
+      if (listOfWords.length === 0) {    
+        return null;
+         }   
+         else { 
+             return (sumWordsLength(listOfWords)/listOfWords.length)  
+             } 
+            }
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(avgArray) {
+  if (avgArray.length === 0){
+    return null;
+  }
+  
```

Comment:
For this iteration you need a loop that will run through all the elements of the array, and in each element you need to check the data type of that element and depending on the data type you do different operations